packages/core/admin/server/bootstrap.js

```diff
@@ -73,6 +90,7 @@ module.exports = async () => {
   await userService.displayWarningIfUsersDontHaveRole();
 
   await syncAuthSettings();
+  await syncAPITokensPermissions();
```

Comment:
Now that I see this, I'm not sure I like the idea of running it on startup (or ever, unless it's an explicit cleanup command). If something goes wrong registering actions, we wipe out all token permissions. I think it might be better to only clean them up on update (like we're already doing).

Reply:
Perhaps the admin API can filter invalid permissions out in find/findOne so we never see the invalid permissions

Reply:
The reasoning behind this being run on startup is to keep the database as the source of truth for the application.
Even if the code in the backend is not using the permissions, it makes no sense to return obsolete actions to the frontend (or store them). it can also lead to confusion for plugins or other services trying to gather metrics or aggregate data from the permissions. One might try to determine if there is at least one permission attached to an API token. To do so, they query the table and see that there are indeed some permissions... even if they're all obsolete :confused:

Reply:
To give you a bit more context, this cleanup/synchronization step that we're doing here is quite common in the application as we try to keep the database clean & up-to-date so that it can serve as a source of truth. Thus, you can find the same logic for RBAC (and the roles/permissions), for the schemas (stored in the core store), etc...

Reply:
> Perhaps the admin API can filter invalid permissions out in find/findOne so we never see the invalid permissions
That would add a patch on one endpoint, but the corrupted data remains available for every other API (entity service, entity manager/repository, etc...)

Reply:
Ok, just my paranoia then 😄